registration/registration_tests.py

```diff
@@ -211,10 +214,68 @@ def test_registration__21(db, user):
     assert not testuser.has_perm('registration.verified')
 
 
+def test_registration__22(browser, user, mail_outbox):
+    """User can download a plain application form."""
+    register_with_pwd(browser, user, mail_outbox)
+    _log_in(browser, user['email'], PWD)
+    browser.open(APPLY_URL)
+    browser.getControl('Apply').click()
+    # TODO don't checks content of pdf
+    assert browser.headers['Content-Type'] == 'application/pdf'
+
+
+def test_registration__23(browser, user, mail_outbox):
+    """User can download an automatically created application form."""
+    register_with_pwd(browser, user, mail_outbox)
+    _log_in(browser, user['email'], PWD)
+    browser.open(APPLY_URL)
+    browser.getControl('application form').click()
+    # TODO don't checks content of pdf
+    assert browser.headers['Content-Type'] == 'application/pdf'
+
+
+def test_registration__24(browser, user):
+    """User without a profile can not create an application form."""
+    testuser = User.objects.create_user(email=user['email'], password=PWD)
+    testuser.save()
+    _log_in(browser, user['email'], PWD)
+    with pytest.raises(HTTPError, match=r'.*500.*'):
+        browser.open(APPLY_URL)
+
+
+def test_registration__25(browser, user, mail_outbox):
```

Comment:
We try to keep the following convention for test names.
`test_{filenameOfClass/Function}__[{ClassName}__][{method_name}__]{XX}():`
This helps to get a hint of the implementation and is also handy in case of a removal of code. 
https://projects.gocept.com/projects/infrastructure/wiki/Namenskonventionen_f%C3%BCr_Tests

This test in particular is not testing the registration but the editability of the `user_data_view`. The "registration" or rather creation of a user should be done programmatically e.g. in a fixture instead of using the testbrowser. This should be faster and is better design, as the creation of a user should be independent of the actual registration form.

Reply:
Maybe it would be better to do this in a separate PR. I added an [issue](https://github.com/gocept/ok_tools/issues/21#issue-1318243443) for this. I think if I refactor the tests in this PR it would lead to huge merge conflicts with the other open PRs